mitodl/open-discussions
# add livestream widget to home page

#### Pre-Flight checklist



#### What are the relevant tickets?

closes #1767 

#### What's this PR do?

This basically adds a little widget to the homescreen which will showcase an event which is currently being livestreamed by MIT video's livestream account.

This involves:

- adding code on the backend for storing API credentials and using them to get information about the relevant account
- adding an API which makes that information available to the frontend
- adding to the frontend a new component which makes a request to that API and, if there is a currently-live event, uses an iframe to embed the event (allowing the user to watch it right there).

#### How should this be manually tested?

message me for configuration stuff. then test that, with things configured, the homepage works normally. you should see a new request being made to the livestream API on the homepage, which should not result in anything being displayed.

in order to do an end-to-end test we need to have a currently live event. I have my phone set up so I can do this on our test account, so if you're testing it just lmk and I'll set up an event so it will come back as a "live" event and you can test the whole thing.


#### Screenshots (if appropriate)

here's what it looks like when there is an event:

![basicdisplay](https://user-images.githubusercontent.com/6207644/53350204-a567ee80-38ec-11e9-8686-43b4a12d193d.png)
